optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/thread/MoveThreadRunner.java

```diff
@@ -46,6 +46,7 @@
     private final boolean assertShadowVariablesAreNotStaleAfterStep;
 
     private InnerScoreDirector<Solution_> scoreDirector = null;
+    private volatile long calculationCount = -1L;
```

Comment:
If multiple threads can write to this variable, `AtomicLong` should be used instead of `volatile`.

Reply:
true. long's are not guaranteed to have atomic read/writes, unlike int's.
only 1 thread writes to it, but that doesn't excuse it.

Reply:
Done